src/main/java/GameMonitor.java

```diff
@@ -1,33 +1,53 @@
 public class GameMonitor {
 
-    Players firstPlayer;
-    Players secondPlayer;
+    Player firstPlayer;
+    Player secondPlayer;
     Game match;
 
-    public GameMonitor () {
-        this.firstPlayer = new Players();
-        this.secondPlayer = new Players();
+    public GameMonitor() {
+        this.firstPlayer = new Player();
+        this.secondPlayer = new Player();
         this.match = new Game(firstPlayer, secondPlayer);
+
     }
 
-    public void buttonPress(Message message) {
+    public Player getPlayer(Message message) {
 
-         if (message.getButtonId().equals("1")) {
-             match.updateScoreOfPlayer(firstPlayer, message.getActionType());
-             if (match.hasSomebodyWon()) {
-                 match.updateMatchScoreOfPlayers();
-             }
+        if (message.getButtonId().equals("1") && match.isSetScoreEven(firstPlayer, secondPlayer)) {
+            return  this.firstPlayer;
+        } else if (message.getButtonId().equals("2") && match.isSetScoreEven(firstPlayer, secondPlayer)) {
+            return this.secondPlayer;
+        } else if (message.getButtonId().equals("1")) {
+            return this.secondPlayer;
+        } else {
+            return this.firstPlayer;
+        }
 
-        } else if (message.getButtonId().equals("2")) {
-            match.updateScoreOfPlayer(secondPlayer, message.getActionType());
-            if (match.hasSomebodyWon()) {
-                match.updateMatchScoreOfPlayers();
-            }
+    }
 
+    public void buttonPress(Message message) {
+        Player actingPlayer = this.getPlayer(message);
```

Comment:
Nitpick: do you mean `activePlayer`? 🤔

Reply:
I think "acting" is a better designation. The important thing is to know which player is acting (which player has made a point or wants to update the score) and both are active, aren't they?